WIP: Rewrite Debugger to use dbgeng.dll

centlakestefan opened the pull request:
Here is some work in progress code for the Debugger part using the dbgeng.dll module from Microsoft.

Using this code I can run OpenCppCovarage in a docker container.

dbgeng.dll is not redistributable from Micorosoft but it is already installed in mcr.microsoft.com/windows/servercore:ltsc2019

I have only tested using the x64 build.

centlakestefan committed:
WIP: Rewrite Debugger to use dbgeng.dll.

centlakestefan committed:
win32 build fix

centlakestefan committed:
Make sure to get an initial breakpoint to prevent crashes

pontzio commented:
Hi! I am really interested in such a feature. Do you know if they expect to include it in the "master" branch? Otherwise, would it be possible for you to share the executable?

centlakestefan committed:
Update continuous-integration-workflow.yml

Enable manual workflow run

centlakestefan committed:
Update continuous-integration-workflow.yml

New setup-msbuild action version

centlakestefan committed:
Added Windows Installer XML setup project

centlakestefan committed:
Added release workflow

centlakestefan committed:
Fixed tests and dbgeng code to pass all

centlakestefan committed:
Upload artifact at the end of the release workflow

centlakestefan committed:
Fixed one more test-case to pass

centlakestefan committed:
Changed workflows to use latest version of checkout action

centlakestefan committed:
Added missing DLL to MSI installer

centlakestefan commented:
I have now added workflows to unit-test and build a release version to my fork. I also added a very simple MSI package to distribute all the files. Binaries are available under Actions/Release build/Artifacts/MSI.

pontzio commented:
Thank you very much! I have tried it and works perfectly! :)

smithadg commented:
Hi @centlakestefan I cannot see artifacts in release build. https://github.com/centlakestefan/OpenCppCoverage/actions/runs/7281327966 or from https://github.com/centlakestefan/OpenCppCoverage/releases
Where can I download from?

centlakestefan commented:
@smithadg
> Hi @centlakestefan I cannot see artifacts in release build. 

I can see the MSI artifact on the actions page (your first link). I do not know why you can't download it. 
Since the workflow artifacts will be removed after some time I have created a wip release. It should be available using your second link.

smithadg commented:
> @smithadg
> 
> > Hi @centlakestefan I cannot see artifacts in release build.
> 
> I can see the MSI artifact on the actions page (your first link). I do not know why you can't download it. Since the workflow artifacts will be removed after some time I have created a wip release. It should be available using your second link.

Thanks a lot. I am able to download.

OpenCppCoverage commented:
Thank you for preparing this pull request and contributing your time to the project.

The project is now being formally retired and the repository will be archived. I am therefore closing all remaining pull requests, as I am no longer able to review, merge, or maintain proposed changes.

This administrative closure is not a judgment on the quality or usefulness of the contribution. The pull request and its discussion will remain available, and anyone continuing the project in a fork is welcome to reuse or reconsider the proposed changes, subject to the project’s licence.

Thank you again for your contribution.
